lib/instrumentation/decorator.ts

```diff
@@ -0,0 +1,46 @@
+import type { Attributes, SpanKind } from '@opentelemetry/api';
+import { Decorator, decorate } from '../util/decorator';
+import { instrument as instrumentFunc } from '.';
+
+/**
+ * The instrumentation decorator parameters.
+ */
+interface SpanParameters {
+  /**
+   * The name of the span
+   */
+  name: string;
+
+  /**
+   * Attributes which should be added to the span
+   */
+  attributes?: Attributes;
+
+  /**
+   * Should this span be added to the root span or to the current active span
+   */
+  ignoreParentSpan?: boolean;
+
+  /**
+   * Type of span this represents. Default: SpanKind.Internal
+   */
+  kind?: SpanKind;
+}
+
+/**
+ * instruments a decorated method.
+ */
+export function instrument<T>({
+  name,
+  attributes,
+  ignoreParentSpan,
+  kind,
+}: SpanParameters): Decorator<T> {
+  return decorate(async ({ callback }) => {
+    return await instrumentFunc(name, async () => await callback(), {
```

Comment:
```suggestion
    return await instrumentFunc(name, callback, {
```
why another lamda function wrapper?

Reply:
Callbacks are not async functions an therefore the linting fails with 
```
TS2345: Argument of type '() => unknown' is not assignable to parameter of type '() => Promise<unknown>'.   Type 'unknown' is not assignable to type 'Promise<unknown>'
```